components/remote_settings/src/service.rs

```diff
@@ -63,7 +63,12 @@ impl RemoteSettingsService {
         context: Option<RemoteSettingsContext>,
     ) -> Result<Arc<RemoteSettingsClient>> {
         let mut inner = self.inner.lock();
-        let storage = Storage::new(inner.storage_dir.join(format!("{collection_name}.sql")))?;
+        // Allow using in-memory databases for testing of external crates.
+        let storage = if inner.storage_dir == ":memory:" {
+            Storage::new(inner.storage_dir.clone())?
```

Comment:
Maybe this should be Something like `file:remote-settings-{collection-name}?mode=memory&cache=shared`.  That way clients for the same collection will get the same in-memory db.